### ISSUE TITLE
Tokenization issue after upgrading to T2T 1.2.4

### ISSUE THREAD

**Issue by fstahlberg** [tensorflow/tensor2tensor#333]
My t2t-decoder started to insert a blank symbol after each character after upgrading to T2T 1.2.4:

>  t2t-decoder \
>    --data_dir=t2t_data/ \
>    --problems=translate_ende_wmt32k \
>    --model=transformer \
>    --hparams_set=transformer_base
>  (...)
>  INFO:tensorflow:Inference results INPUT: Airport building evacuated
>  INFO:tensorflow:Inference results OUTPUT: F l u g h a f e n g e b \<C3\> \<A4\> u d e   e v a k u i e r t

If I remove the blanks afterwards with

>  cat decodes | sed 's/   /___/g' | tr -d ' ' | sed 's/___/ /g'

the output looks okay again. I have generated the data and trained the model with T2T 1.2.3.

**Comment by fstahlberg** [comment]
So apparently [this commit](https://github.com/tensorflow/tensor2tensor/commit/ed7862c95a42323b775573a7b508409a7d167afc#diff-189703c3c1f690231282073cea1cec41) is the problem. Is there a particular reason for doing this or was that pushed by mistake? And why not just do

```
decoded_outputs = targets_vocab.decode(_save_until_eos(outputs.flatten()))
if targets is not None:
  decoded_targets = targets_vocab.decode(_save_until_eos(targets.flatten()))
```

**Comment by mudong0419** [comment]
I have the same problem, and I think it is a bug.

**Comment by fstahlberg** [comment]
Fixed with 1.2.5, thanks
